ubccr/xdmod
# Acl config sync

## Description
Ensure that `acl-config` adds support for both the removal and modification of configuration files that populate the tables that it manages as well as addition.  

## Motivation and Context
So that we can truly manage the state of the acl tables w/ config file modifications.

## Tests performed
Manual Tests: 

**Test 1:**
- Start an Open XDMoD docker container and connect
- Start the services provided by the docker container ( `~/bin/services start` )
- Add a new acl to the 'normaluser' user
  - Could be via directly adding a record in `moddb.user_acls` or via the internal dashboard
- Perform an source upgrade
- Verify that the newly added acl relation still exists
  - By either directly via the db or via the internal dashboard

**Test 2:**
- Start an Open XDMoD docker container and connect
- Start the container services ( `~/bin/services start` )
- Add a new acl to the system
  - Either modify roles.json and run `acl-config` or manually add a record to `moddb.acls`
- Perform a source upgrade -or- if roles.json was modified in the previous step then revert the file and run `acl-config` again 
- Verify that the newly added acl is not present
  - either via db query or internal dashboard

**NOTE: I also ran all of the unit, integration and component tests for each Test ( which passed )**
## Types of changes


## Checklist:
